autopromote/autopromote.py

```diff
@@ -252,9 +252,19 @@ def promote_pkg(current_plist, path):
 
     logger.info(f"Considering package {fullname}")
 
-    if name in denylist or (allowlist and name not in allowlist):
-        logger.warn(f"Skipping {fullname}: excluded by allowlist/denylist")
-        return promoted, result
+    # If the name and name-version are in the deny list AND name is not in the allow list
+    if (name in denylist or name + "-" + version in denylist) and (allowlist and name not in allowlist):
```

Comment:
This logic needs to be fixed. Specifically the `and` - right now it is `if <denylist applied>` _and_ `if <allow list exists and allowlist applied>`. Should be `or`. Simplified, the logic is "if this item is denied _or_ there exists an allow list and this item isn't on it". Allowlist takes precedence if it exists.

Reply:
Ah yes, totally missed the `not`, told you I was working late and to check!